app/kumactl/cmd/delete/delete_meshes_test.go

```diff
@@ -0,0 +1,155 @@
+package delete_test
+
+import (
+	"bytes"
+	"context"
+	"path/filepath"
+	"time"
+
+	"github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/app/kumactl/cmd"
+	config_proto "github.com/Kong/kuma/pkg/config/app/kumactl/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/Kong/kuma/pkg/core/resources/model"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	memory_resources "github.com/Kong/kuma/pkg/plugins/resources/memory"
+	test_model "github.com/Kong/kuma/pkg/test/resources/model"
+
+	kumactl_cmd "github.com/Kong/kuma/app/kumactl/pkg/cmd"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+var _ = Describe("kumactl delete mesh", func() {
+
+	sampleMeshes := []*mesh.MeshResource{
+		{
+			Spec: v1alpha1.Mesh{
+				Mtls: &v1alpha1.Mesh_Mtls{
+					Enabled: true,
+					Ca: &v1alpha1.CertificateAuthority{
+						Type: &v1alpha1.CertificateAuthority_Builtin_{
+							Builtin: &v1alpha1.CertificateAuthority_Builtin{},
+						},
+					},
+				},
+				Logging: &v1alpha1.Logging{
+					AccessLogs: &v1alpha1.Logging_AccessLogs{
+						Enabled:  true,
+						FilePath: "/tmp/access.log",
+					},
+				},
+			},
+			Meta: &test_model.ResourceMeta{
+				Mesh:      "mesh1",
+				Name:      "mesh1",
+				Namespace: "default",
+			},
+		},
+		{
+			Spec: v1alpha1.Mesh{
+				Mtls: &v1alpha1.Mesh_Mtls{
+					Enabled: false,
+					Ca: &v1alpha1.CertificateAuthority{
+						Type: &v1alpha1.CertificateAuthority_Builtin_{
+							Builtin: &v1alpha1.CertificateAuthority_Builtin{},
+						},
+					},
+				},
+			},
+			Meta: &test_model.ResourceMeta{
+				Mesh:      "mesh2",
+				Name:      "mesh2",
+				Namespace: "default",
+			},
+		},
+	}
+
+	Describe("DeleteMeshCommand", func() {
+
+		var rootCtx *kumactl_cmd.RootContext
+		var rootCmd *cobra.Command
+		var buf *bytes.Buffer
+		var store core_store.ResourceStore
+
+		BeforeEach(func() {
+			// setup
+			rootCtx = &kumactl_cmd.RootContext{
+				Runtime: kumactl_cmd.RootRuntime{
+					Now: func() time.Time { return time.Now() },
+					NewResourceStore: func(*config_proto.ControlPlaneCoordinates_ApiServer) (core_store.ResourceStore, error) {
+						return store, nil
+					},
+				},
+			}
+
+			store = memory_resources.NewStore()
+
+			for _, ds := range sampleMeshes {
+				key := core_model.ResourceKey{
```

Comment:
You can use `pkg/core/resources/model.MetaToResourceKey(..)` func to convert `ds.Meta` into `ResourceKey`